packages/catalog-process/src/model/domain/apiConverter.ts

```diff
@@ -0,0 +1,166 @@
+    .with(descriptorState.archived, () => "ARCHIVED")
+    .exhaustive();
+}
+
+export function apiDescriptorStateToDescriptorState(
+  input: ApiEServiceDescriptorState
+): DescriptorState {
+  return match<ApiEServiceDescriptorState, DescriptorState>(input)
+    .with("DRAFT", () => descriptorState.draft)
+    .with("PUBLISHED", () => descriptorState.published)
+    .with("SUSPENDED", () => descriptorState.suspended)
+    .with("DEPRECATED", () => descriptorState.deprecated)
+    .with("ARCHIVED", () => descriptorState.archived)
+    .exhaustive();
+}
+
+export function agreementApprovalPolicyToApiAgreementApprovalPolicy(
+  input: AgreementApprovalPolicy | undefined
+): ApiAgreementApprovalPolicy {
+  return match<AgreementApprovalPolicy | undefined, ApiAgreementApprovalPolicy>(
+    input
+  )
+    .with(agreementApprovalPolicy.automatic, () => "AUTOMATIC")
+    .with(agreementApprovalPolicy.manual, () => "MANUAL")
+    .otherwise(() => "AUTOMATIC");
+}
+
+export function agreementStateToApiAgreementState(
+  input: PersistentAgreementState
+): ApiAgreementState {
+  return match<PersistentAgreementState, ApiAgreementState>(input)
+    .with(persistentAgreementState.pending, () => "PENDING")
+    .with(persistentAgreementState.rejected, () => "REJECTED")
+    .with(persistentAgreementState.active, () => "ACTIVE")
+    .with(persistentAgreementState.suspended, () => "SUSPENDED")
+    .with(persistentAgreementState.archived, () => "ARCHIVED")
+    .with(persistentAgreementState.draft, () => "DRAFT")
+    .with(
+      persistentAgreementState.missingCertifiedAttributes,
+      () => "MISSING_CERTIFIED_ATTRIBUTES"
+    )
+    .exhaustive();
+}
+
+export function apiAgreementStateToAgreementState(
+  input: ApiAgreementState
+): PersistentAgreementState {
+  return match<ApiAgreementState, PersistentAgreementState>(input)
+    .with("PENDING", () => persistentAgreementState.pending)
+    .with("REJECTED", () => persistentAgreementState.rejected)
+    .with("ACTIVE", () => persistentAgreementState.active)
+    .with("SUSPENDED", () => persistentAgreementState.suspended)
+    .with("ARCHIVED", () => persistentAgreementState.archived)
+    .with("DRAFT", () => persistentAgreementState.draft)
+    .with(
+      "MISSING_CERTIFIED_ATTRIBUTES",
+      () => persistentAgreementState.missingCertifiedAttributes
+    )
+    .exhaustive();
+}
+
+export const eServiceToApiEService = (
+  eService: EService
+): z.infer<typeof api.schemas.EService> => {
+  const mapAttribute = (
+    a: Attribute
+  ):
+    | {
+        single: {
+          id: string;
+          explicitAttributeVerification: boolean;
+        };
+      }
+    | {
+        group: Array<{
+          id: string;
+          explicitAttributeVerification: boolean;
+        }>;
+      } =>
+    match(a)
+      .with({ type: "SingleAttribute" }, (a) => ({
+        single: a.id,
+      }))
+      .with({ type: "GroupAttribute" }, (a) => ({
+        group: a.ids,
+      }))
+      .exhaustive();
```

Comment:
this assumes that the read model stores attributes with a discriminator field `type` that is either `SingleAttribute` or `GroupAttribute`

however, the serialization code [here](https://github.com/pagopa/interop-be-catalog-management/blob/1.0.x/models/src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/JsonFormats.scala/#L67) does not appear to use a discriminating field: it just assumes it's a group attribute if it has an `ids` field and a single attribute otherwise

Reply:
tracked as https://buildo.atlassian.net/browse/PPA-28